cpp/src/traversal/tsp.cu

```diff
@@ -50,139 +52,160 @@ TSP::TSP(raft::handle_t &handle,
-  for (int b = 0; b < num_restart_batches; ++b) {
-    reset<<<1, 1, 0, stream_>>>(mylock_, best_tour_, climbs_);
-    CHECK_CUDA(stream_);
+  for (auto batch = 0; batch < num_restart_batches; ++batch) {
+    reset_batch();
+    if (batch == num_restart_batches - 1) restart_batch_ = restart_resid;
 
-    if (b == num_restart_batches - 1) restart_batch_ = restart_resid;
-
-    search_solution<<<restart_batch_, threads, sizeof(int) * threads, stream_>>>(mylock_,
-                                                                                 best_tour_,
-                                                                                 vtx_ptr_,
-                                                                                 beam_search_,
-                                                                                 k_,
-                                                                                 nodes_,
-                                                                                 neighbors_,
-                                                                                 x_pos_,
-                                                                                 y_pos_,
-                                                                                 work_,
-                                                                                 nstart_,
-                                                                                 times.data().get(),
-                                                                                 climbs_,
-                                                                                 threads);
+    {
+      auto timer = create_timer("initial_sol");
+      get_initial_solution(batch);
+    }
 
-    CHECK_CUDA(stream_);
-    cudaDeviceSynchronize();
+    {
+      auto timer = create_timer("search_sol");
+      search_solution<<<restart_batch_,
+                        best_thread_num_,
+                        sizeof(int) * best_thread_num_,
+                        stream_>>>(
+        results_, mylock_, vtx_ptr_, beam_search_, k_, nodes_, x_pos_, y_pos_, work_, nstart_);
+      CHECK_CUDA(stream_);
+    }
+
+    {
+      auto timer = create_timer("optimal_tour");
+      get_optimal_tour<<<restart_batch_,
+                         best_thread_num_,
+                         sizeof(int) * best_thread_num_,
+                         stream_>>>(results_, mylock_, work_, nodes_);
+      CHECK_CUDA(stream_);
+    }
 
-    CUDA_TRY(cudaMemcpy(&best, best_tour_, sizeof(int), cudaMemcpyDeviceToHost));
     cudaDeviceSynchronize();
+    best = best_cost_scalar_.value(stream_);
+
     if (verbose_) std::cout << "Best reported by kernel = " << best << "\n";
 
     if (best < global_best) {
       global_best = best;
-      CUDA_TRY(cudaMemcpyFromSymbol(&soln, best_soln, sizeof(void *)));
-      cudaDeviceSynchronize();
-      CUDA_TRY(cudaMemcpyFromSymbol(&route_sol, best_route, sizeof(void *)));
+      cudaMemcpy(
```

Comment:
Why was `CUDA_TRY()` dropped; i.e., not enclosing the `cudamemcpy()`'s anymore?

Reply:
Thanks, updated here: https://github.com/rapidsai/cugraph/pull/1480/commits/b7ae073555806fe62fdb52cf2550080e45ab64e8

Reply:
consider `raft::update_host` unless there's a reason not to use it.

Reply:
Changed here: https://github.com/rapidsai/cugraph/pull/1480/commits/03f2d2f90d644855ba2e777597235452123add8c